apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/ProjectMetricsResourceTest.java

```diff
@@ -398,67 +412,35 @@ void happyPath(TimeInterval interval) {
-            assertThat(response.metricType()).isEqualTo(MetricType.FEEDBACK_SCORES);
-            assertThat(response.interval()).isEqualTo(interval);
-            assertThat(response.results()).hasSize(names.size());
-
-            assertThat(response.results()).hasSize(expected.size());
-            assertThat(response.results())
-                    .usingRecursiveComparison()
-                    .withComparatorForType(this::bigDecimalInDelta, BigDecimal.class)
-                    .ignoringCollectionOrder()
-                    .isEqualTo(expected);
+                    .build(), marker, names, BigDecimal.class, scoresMinus3, scoresMinus1, scores);
         }
 
-        private List<ProjectMetricResponse.Results<BigDecimal>> createExpectedFeedbackScores(
-                Instant marker, TimeInterval interval, List<String> names, Map<String, BigDecimal> scoresMinus3,
-                Map<String, BigDecimal> scoresMinus1, Map<String, BigDecimal> scores) {
-            return names.stream()
-                    .map(name -> {
-                        var expectedFeedbackScores = Arrays.asList(
-                                null,
-                                scoresMinus3.get(name),
-                                null,
-                                scoresMinus1.get(name),
-                                scores.get(name));
-
-                        return ProjectMetricResponse.Results.<BigDecimal>builder()
-                                .name(name)
-                                .data(IntStream.range(0, expectedFeedbackScores.size())
-                                        .mapToObj(i -> DataPoint.<BigDecimal>builder()
-                                                .time(subtract(marker, 4 - i, interval))
-                                                .value(expectedFeedbackScores.get(i)).build())
-                                        .toList())
-                                .build();
-                    }).toList();
-        }
-
-        private int bigDecimalInDelta(BigDecimal number, BigDecimal number2) {
-            if (number == null) {
-                return number2 == null ? 0 : 1;
-            }
-            if (number2 == null) {
-                return -1;
-            }
+        @ParameterizedTest
+        @EnumSource(TimeInterval.class)
+        void emptyData(TimeInterval interval) {
+            // setup
+            mockTargetWorkspace();
 
-            var delta = new BigDecimal(".000001");
-            if (number.subtract(number2).abs().compareTo(delta) < 0) {
-                return 0;
-            }
+            Instant marker = getIntervalStart(interval);
+            String projectName = RandomStringUtils.randomAlphabetic(10);
+            var projectId = projectResourceClient.createProject(projectName, API_KEY, WORKSPACE_NAME);
+            Map<String, BigDecimal> empty = new HashMap<>() {
+                {
+                    put("", null);
```

Comment:
Can we replace the empty map?

Reply:
With what? This is what the FE expects

Reply:
Yeah, avoiding returning "": null in the response

Reply:
Sorry, I thought because `request.interval()`, but I see this is not passed directly

Reply:
The response will look like this:
```json
{
  "project_id": "01936d62-a28b-788b-a0aa-c13271bf2166",
  "metric_type": "TOKEN_USAGE",
  "interval": "HOURLY",
  "results": [
    {
      "name": "",
      "data": [
        {
          "time": "2024-11-27T07:00:00Z",
          "value": null
        },
        {
          "time": "2024-11-27T08:00:00Z",
          "value": null
        },
        {
          "time": "2024-11-27T09:00:00Z",
          "value": null
        },
        {
          "time": "2024-11-27T10:00:00Z",
          "value": null
        },
        {
          "time": "2024-11-27T11:00:00Z",
          "value": null
        }
      ]
    }
  ]
}
```
And this is expected by the FE.

Reply:
Cannot we remove such a row from the array?

Reply:
No, the FE expects it in order to plot the empty X axis. Removing it from the array will result in an empty array.